native/src/main/java/io/ballerina/lib/data/jsonpath/BJsonProvider.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.lib.data.jsonpath;
+
+import com.jayway.jsonpath.InvalidJsonException;
+import com.jayway.jsonpath.JsonPathException;
+import com.jayway.jsonpath.spi.json.AbstractJsonProvider;
+import io.ballerina.runtime.api.PredefinedTypes;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.utils.JsonUtils;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.io.InputStream;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Provides Ballerina specific function implementations of json-path.
+ */
+public class BJsonProvider extends AbstractJsonProvider {
+
+    /**
+     * Parse the given json string.
+     *
+     * @param json json string to parse
+     * @return Object representation of json
+     * @throws InvalidJsonException
+     */
+    @Override
+    public Object parse(String json) throws InvalidJsonException {
+        return JsonUtils.parse(json);
+    }
+
+    /**
+     * Parse the given json bytes in UTF-8 encoding.
+     *
+     * @param json json bytes to parse
+     * @return Object representation of json
+     * @throws InvalidJsonException
+     */
+    @Override
+    public Object parse(byte[] json) throws InvalidJsonException {
+        return super.parse(json);
```

Comment:
Can you add a test to cover this scenario?

Reply:
Currently we are supporting json types only for the reading scenarios.
These methods are override for completeness and future improvements